catchup/service.go

```diff
@@ -73,6 +73,10 @@ type Service struct {
 	auth            BlockAuthenticator
 	parallelBlocks  uint64
 	deadlineTimeout time.Duration
+	// catchpointWriting defines whether we've ran into a state where the ledger is currently busy writing the
+	// catchpoint file. If so, we want to pospone all the catchup process until the catchpoint file writing is complete,
+	// and resume from there without stopping the catchup timer.
+	catchpointWriting bool
```

Comment:
I want to propose couple of different names for catchpointWriting:
syncInterruptedWaitingCatchpointWriting
syncWaitingForCatchpointWriting

catchpointWriting is lacking context, and I found it difficult to understand the logic without this context.